### ISSUE TITLE
[Feature] Group channel invitations should be true accept/decline invitations, not forced-join

### ISSUE THREAD

**Issue by axzhandul** [oxen-io/session-ios#917]
### Is there an existing request for feature?

- [X] I have searched the existing issues

### What feature would you like?

Please forgive the submission of this issue here, as I imagine it effects the entire platform (all clients and server). However, I looked through the repos and wasn’t sure which one to post on, and since I discovered this using the iOS client, I’m posting here. If this belongs as an issue on a different repo, hopefully an admin can move it. To the issue….

Presently, it appears that others can “invite” you to a group channel, but that the nature of this isn’t a true invitation, where you are able to accept or decline the invitation, but rather you are automatically joined to the channel by virtue of nothing but the invite of the other. This gives others the ability to control your own personal associations. By virtue of the way Display Name data is handled, it also gives others the ability to burn the privacy of your Session ID to the membership of any channel they invite you to, without your consent. 

Aside from the obvious, this has many other life metaphors, and in none of them is it a good idea for others to dictate whether you are associated or not. Membership in any group / channel should only be at the consent / confirmation of the owner of the Session ID. 

I would suggest that an invitation give the invited party the ability to preview the discussion on the channel (without joining), and either accept / decline the invitation, resulting in either joining or not joining, respectively.

**Comment by KeeJef** [comment]
Yeah we are addressing this in our closed group redesign, in the new closed group design invites flow through the message requests system so you have to approve/deny invitations to groups, this will work in the same way as 1-1 messages
